src/Message/ExpressAuthorizeRequest.php

```diff
@@ -34,6 +91,36 @@ public function getData()
         $data['LOCALECODE'] = $this->getLocaleCode();
         $data['CUSTOMERSERVICENUMBER'] = $this->getCustomerServiceNumber();
 
+        $callback = $this->getCallback();
+
+        if (!empty($callback)) {
+            $data['CALLBACK'] = $callback;
+            // callback timeout MUST be included and > 0
+            $timeout = $this->getCallbackTimeout();
+
+            $data['CALLBACKTIMEOUT'] = $timeout > 0 ? $timeout : self::DEFAULT_CALLBACK_TIMEOUT;
+
+            // if you're using a callback you MUST set shipping option(s)
+            $shippingOptions = $this->getShippingOptions();
+
+            if (!empty($shippingOptions)) {
+                foreach ($shippingOptions as $index => $shipping) {
+                    $name = $shipping['name'];
```

Comment:
Agreed on this, there's no precedent in the project for making a shipping options wrapper class and passing an array of instances through though. That would be my preferred way of verifying content. All the required parameters would be on its constructor and the optional label would simply have a setter.

End users would do something like this:

```
$request->setShippingOptions([
  new PaypalShippingOptions('1st Class', 2.50, true),
  new PaypalShippingOptions('2nd Class', 3.50),
]);
```
The setShippingOptions would check to see if it least one of them has been set as default. 

More robust than arrays and self-documenting in a decent IDE. Any thoughts or happy to go with an array & key validation throws exception?